src/DotNetWorker.Core/FunctionMetadata/IFunctionMetadata.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System.Collections.Generic;
+
+namespace Microsoft.Azure.Functions.Worker.Core.FunctionMetadata
+{
+    /// <summary>
+    /// Public interface that represents properties in function metadata.
+    /// </summary>
+    public interface IFunctionMetadata
+    {
+        /// <summary>
+        /// Unique function id.
+        /// </summary>
+        string FunctionId { get; set; }
+
+        /// <summary>
+        /// If function is a proxy function return true, else return false.
+        /// </summary>
+        bool IsProxy { get; set; }
+
+        /// <summary>
+        /// Language that the function is written in.
+        /// </summary>
+        string Language { get; set; }
+
+        /// <summary>
+        /// If managed dependency is enabled return true, else return false.
+        /// </summary>
+        bool ManagedDependencyEnabled { get; set; }
```

Comment:
nit - should we make it `IsManagedDependencyEnabled` to keep it consistent with other bool variables like `IsProxy`. Please ignore if it does not make sense with the context it is being used.

Reply:
The fields here are named to be converted into RpcFunctionMetadata easily w/ json serializiation/deserialization. [See corresponding property in proto file](https://github.com/Azure/azure-functions-dotnet-worker/blob/fcda116a6fae1025d5de3fb6663f5d1d973a5941/src/DotNetWorker.Grpc/protobuf/src/proto/FunctionRpc.proto#L323). The proto file uses `IsProxy` but for this property there's no "is" in front :(

Reply:
Sure. Got it.